README.md

```diff
@@ -38,25 +38,60 @@
 
 <br/>
 
-## Environment Variables
-
-| Variable name                         | Description                   |
-|---------------------------------------|-------------------------------|
-| BUX_ENVIRONMENT                       | development or production     |
-| BUX_AUTHENTICATION\*\*ADMIN_KEY       | [your-admin-xpriv]            |
-| BUX_AUTHENTICATION\*\*REQUIRE_SIGNING | 1 or 0                        |
-| BUX_DATASTORE\*\*ENGINE               | mongodb, sqlite, mysql        |
-| BUX_MONGODB\*\*URI                    | mongodb://localhost:27017/bux |
-| BUX_MONGODB\*\*DATABASE_NAME          | bux                           |
-| BUX_DATASTORE\*\*AUTO_MIGRATE         | 1 or 0                        |
-| BUX_CACHE\*\*ENGINE                   | freecache or redis            |
-| BUX_REDIS\*\*URL                      | redis://localhost:6379        |
-| BUX_MONITOR\*\*ENABLED                | 1 or 0                        |
-| BUX_GRAPHQL\*\*ENABLED                | 1 or 0                        |
-| BUX_DEBUG                             | 1 or 0                        |
-| BUX_NEW_RELIC\*\*ENABLED              | 1 or 0                        |
-| BUX_NEW_RELIC\*\*DOMAIN_NAME          | bux-nyc1.bux.ninja            |
-| BUX_NEW_RELIC\*\*LICENSE_KEY          | [your-key]                    |
+## Config Variables
+
+Default config variables can be overridden by (in this order of importance):
+1. Flags (only the ones below)
+2. ENV variables
+3. Config file
+
+#### Flags
+
+Available flags:
+
+```bash
+  -C, --config_file string                       custom config file path
+  -h, --help                                     show help
+  -v, --version                                  show version
+  -d, --dump_config                              dump config to file, specified by config_file (-C) flag
+```
+
+To generate config file with defaults, use the --dump flag, or:
+```bash
+go run ./cmd/server/main.go -d
+```
+
+The default config file path is **project root**, and the default file name is **config.yaml**. This can be overridden by -C flag.
+```bash
+go run ./cmd/server/main.go -C /my/config.json
+```
+
+#### Environment variables
+
+To override any config variable with ENV, use the "BUX\_" prefix with mapstructure annotation path with "_" as a delimiter in all uppercase. Example:
+
+Let's take this fragment of AppConfig from `config/config.go`:
+```go
+type AppConfig struct {
+	Authentication *AuthenticationConfig `json:"auth" mapstructure:"auth"`
+	Cache *CacheConfig `json:"cache" mapstructure:"cache"`
+	Db *DbConfig `json:"db" mapstructure:"db"`
+	Debug bool `json:"debug" mapstructure:"debug"`
+    ...
+}
+
+type AuthenticationConfig struct {
+	AdminKey string `json:"admin_key" mapstructure:"admin_key"`
+	RequireSigning bool `json:"require_signing" mapstructure:"require_signing"`
+	Scheme string `json:"scheme" mapstructure:"scheme"`
+	SigningDisabled bool `json:"signing_disabled" mapstructure:"signing_disabled"`
+}
+```
```

Comment:
Instead of go code, I would rather point to example config file, and write that instruction based on it.

Reply:
Good point, done